packages/core/lib/commands/call/run.js

```diff
@@ -0,0 +1,167 @@
+  const loadConfig = require("../../loadConfig");
+
+  if (options.url && options.network) {
+    const message =
+      "" +
+      "Mutually exclusive options, --url and --network detected!" +
+      OS.EOL +
+      "Please use either --url or --network and try again." +
+      OS.EOL +
+      "See: https://trufflesuite.com/docs/truffle/reference/truffle-commands/#call" +
+      OS.EOL;
+    throw new TruffleError(message);
+  }
+
+  let config = loadConfig(options);
+  await Environment.detect(config);
+
+  const [contractNameOrAddress, functionNameOrSignature, ...args] = config._;
+  let functionEntry, transaction;
+
+  const { encoder, decoder } = config.fetchExternal
+    ? await sourceFromExternal(contractNameOrAddress, config)
+    : await sourceFromLocal(contractNameOrAddress, config);
+
+  ({ abi: functionEntry, tx: transaction } = await encoder.encodeTransaction(
+    functionNameOrSignature,
+    args
+  ));
+
+  if (functionEntry.stateMutability !== "view") {
+    console.log(
+      `WARNING!!! Not a view function\n  ABI stateMutability: ${functionEntry.stateMutability}`
+    );
+  }
+
+  const provider = new Encoder.ProviderAdapter(config.provider);
+  let result;
+  try {
+    result = await provider.request({
+      method: "eth_call",
+      params: [
+        {
+          from: config.from,
+          to: transaction.to,
+          data: transaction.data
+        },
+        config.blockNumber
+      ]
+    });
+  } catch (error) {
+    console.log(`Error Message: ${error.message}\nError Code: ${error.code}`);
+  }
+
+  if (result !== null && result !== undefined) {
+    const [decoding] = await decoder.decodeReturnValue(functionEntry, result);
+
+    for (const { value: result } of decoding.arguments) {
+      console.log(
+        util.inspect(new Codec.Export.ResultInspector(result), {
+          colors: true,
+          depth: null,
+          maxArrayLength: null,
+          breakLength: 79
+        })
+      );
+    }
+  }
+
+  async function sourceFromLocal(contractNameOrAddress, config) {
+    let encoder, decoder;
+    const contractNames = fs
+      .readdirSync(config.contracts_build_directory)
+      .filter(filename => filename.endsWith(".json"))
+      .map(filename => filename.slice(0, -".json".length));
+
+    const contracts = contractNames
+      .map(contractName => ({
+        [contractName]: config.resolver.require(contractName)
+      }))
+      .reduce((a, b) => ({ ...a, ...b }), {});
+
+    const isEmpty = Object.keys(contracts).length === 0;
+    if (isEmpty) {
+      throw new Error(
+        "No artifacts found! Please run `truffle compile` first to compile your contracts"
+      );
+    }
+
+    if (contractNameOrAddress.startsWith("0x")) {
+      const projectInfo = {
+        artifacts: Object.values(contracts)
+      };
+
+      ({ encoder, decoder } = await getEncoderDecoderForContractAddress(
+        contractNameOrAddress,
+        projectInfo
+      ));
+    } else {
```

Comment:
As long as I'm making stylistic notes -- this code could use more comments!  Like, label the two branches here as the address case vs the name case.  I can see that's what you're doing, sure, but explicit labels are still helpful to the reader.  Alternatively, another way of making this more explicit would be to rename the variable in each case, so e.g. in the first case you could open with writing `const address = contractNameOrAddress;`.  But really I just think writing `//address case` would help a fair bit!

Reply:
I guess comments are good enough to understand but renaming the variables in each branch will be a better idea.